lib/macho/exceptions.rb

```diff
@@ -115,15 +119,15 @@ def initialize(lc)
   end
 
   # Raised when a change at an offset is not valid.
-  class OffsetInsertionError < MachOError
+  class OffsetInsertionError < MachOModificationError
     # @param offset [Fixnum] the invalid offset
     def initialize(offset)
       super "Insertion at offset #{offset} is not valid"
     end
   end
 
   # Raised when load commands are too large to fit in the current file.
-  class HeaderPadError < MachOError
+  class HeaderPadError < MachOModificationError
```

Comment:
As pointed out in a main comment, I don't think this should be reclassified. Please carefully review all the exceptions that can be raised by the modification methods and think whether they should be fatal or non-fatal (in non-strict mode).